File: backend/app/schemas/action_items.py
from typing import Literal

from pydantic import BaseModel

Priority = Literal["High", "Medium", "Low"]


class ActionItem(BaseModel):
    person: str | None
    task: str
    deadline: str | None
    priority: Priority


class ActionItemsResult(BaseModel):
    file_id: int
    filename: str
    action_items: list[ActionItem]

File: frontend/src/types/actionItems.ts
export type ActionItemPriority = "High" | "Medium" | "Low";

export interface ActionItem {
  person: string | null;
  task: string;
  deadline: string | null;
  priority: ActionItemPriority;
}

export interface ActionItemsResult {
  file_id: number;
  filename: string;
  action_items: ActionItem[];
}
